connpicker.go

```diff
@@ -0,0 +1,87 @@
+package gocql
+
+import (
+	"sync"
+	"sync/atomic"
+)
+
+type ConnPicker interface {
+	Pick(token) *Conn
+	Put(*Conn)
+	Remove(conn *Conn)
+	Size() (int, int)
+	Close()
+}
+
+type DefaultConnPicker struct {
+	conns []*Conn
+	pos   uint32
+	size  int
+	mu    sync.RWMutex
+}
+
+func NewDefaultConnPicker(size int) *DefaultConnPicker {
+	return &DefaultConnPicker{
+		size: size,
+	}
+}
+
+func (p *DefaultConnPicker) Remove(conn *Conn) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	for i, candidate := range p.conns {
+		if candidate == conn {
+			p.conns[i] = conn
```

Comment:
`p.conns[i] == nil`?

Reply:
`p.conns[i] = nil` perhaps? I think we should maybe eve resize it what do you think?

Reply:
I have set it to nil now. We can change it if you prefer but the rest of the code works guards against nil.

Reply:
nah, nil is fine. That's exactly what I meant :-)